contrib/pkg/createcluster/gcp.go

```diff
@@ -0,0 +1,70 @@
+package createcluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	hivev1gcp "github.com/openshift/hive/pkg/apis/hive/v1alpha1/gcp"
+)
+
+const (
+	gcpCredFile = "osServiceAccount.json"
+)
+
+var _ cloudProvider = (*gcpCloudProvider)(nil)
+
+type gcpCloudProvider struct {
+}
+
+func (p *gcpCloudProvider) generateCredentialsSecret(o *Options) (*corev1.Secret, error) {
+	saFile := filepath.Join(os.Getenv("HOME"), ".gcp", gcpCredFile)
+	log.Infof("Loading gcp service account from: %s", saFile)
+	saFileContents, err := ioutil.ReadFile(saFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.credsSecretName(o),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
```

Comment:
same comment about the format of our secret here. should it match the format that the installer places down into the cluster?

Reply:
Per above probably the same reason, we need a single file with the contents to be picked up by installer.

Reply:
does this work on AWS b/c we're setting environment variables instead of mounting the secret?

Reply:
I think so yes.

Reply:
Yes, for AWS we are using environment variables. For Azure we are using the file.